### PR TITLE
[TRTLLM-15115][test] Prune Starcoder2-3B functional and unit tests

### PR DESCRIPTION
**Pull request by xinhe-nv**
Remove out-of-scope Starcoder2-3B coverage per Test Scope Review 2026.7, including QA/CI list entries, encode/cli harnesses, and 3B-specific unit tests while leaving StarCoder2-7B/15B coverage intact.

## Dev Engineer Review

- Removed StarCoder2-3B accuracy references, harness coverage, fixtures, performance configuration, and unit-test coverage.
- Preserved StarCoder2-7B and StarCoder2-15B coverage.
- Updated QA, A100, and test-duration entries consistently.
- No production-code or API changes were introduced.
- Configuration and test-list changes have valid scope and no apparent format or duplication issues.

## QA Engineer Review

- Removed `TestStarcoder2_3B` CLI accuracy coverage.
- Removed `test_starcoder2_multi_lora`, `STARCODER2_3B_CONFIG`, and related StarCoder2-3B modeling scenarios.
- Removed StarCoder2-3B tokenizer parameterization from the incremental decoding test.
- Removed the related accuracy entries from:
  - `tests/integration/test_lists/qa/llm_function_core.txt`
  - `tests/integration/test_lists/test-db/l0_a100.yml`
- The integration accuracy test was covered by the QA and CI lists. The removed unit-test cases had no separate test-list entries.
- StarCoder2-7B and StarCoder2-15B coverage remains.
- Verdict: **needs follow-up** because two A100 CI runs failed. The subsequent A100X run passed with fail-fast disabled.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

**Review by yingguo-trt — APPROVED** [review]

**Review by StanleySun639 — APPROVED** [review]

File: tests/unittest/_torch/modeling/test_modeling_starcoder2.py [hunk 1]
```diff
@@ -318,109 +297,3 @@ def test_starcoder2_allclose_to_hf(scenario: Scenario) -> None:
     if graph_runner is not None:
         graph_runner.clear()
     kv_cache_manager.shutdown()
-
-
-@torch.no_grad()
-def test_starcoder2_multi_lora(tmp_path) -> None:
```
Comment (QiJune) [inline tests/unittest/_torch/modeling/test_modeling_starcoder2.py:324]: Please keep this test case for multi-lora feature

**Review by QiJune — APPROVED** [review]
LGTM

File: tests/unittest/_torch/modeling/test_modeling_starcoder2.py (hunk 1 above)
Comment (xinhe-nv) [reply tests/unittest/_torch/modeling/test_modeling_starcoder2.py]: @yibinl-nvidia please help to confirm, the file doesn't call in any test list.

Comment (xinhe-nv) [reply tests/unittest/_torch/modeling/test_modeling_starcoder2.py]: discussed with @yibinl-nvidia offline, we need to delete this file.
